azkaban-web-server/src/main/java/azkaban/webapp/servlet/LoginAbstractAzkabanServlet.java

```diff
@@ -345,6 +349,24 @@ protected void doPost(final HttpServletRequest req, final HttpServletResponse re
     }
   }
 
+  /**
+   * Disallows users from logging in by passing their
+   * username and password via the request header where it'd be logged.
+   *
+   * Login attempt is expected if the query string contains the word password and there is a
+   * parameter named password.
+   *
+   * The query string will be empty if the password is passed incorrectly.
+   * The parameter map will contain a password for any login attempt.
+   *
+   * Returns false if req indicates an attempt to login.
+   * Otherwise returns true.
+   */
+  private boolean isIllegalPostRequest(final HttpServletRequest req) {
+    return (req.getParameterMap().containsKey("password") && req.getQueryString() != null &&
```

Comment:
would you mind adding more description in comment on why req.getQueryString().contains("password") is needed?